### DISCUSSION TITLE
Inconsistency between calculated RSS scores and dotplotheatmap

### DISCUSSION THREAD

**Discussion by laurajohanne** [aertslab/scenicplus#410] [Q&A] [answered]
Hi,

I am having som inconsistency when comparing which eRegulon specificity when RSS scores and dotplotheatmap based on "direct_gene_based_AUC" and "direct_region_based_AUC".

It appers that some eRegulons should be specific to a certain cell type but when comparing to the  dotplotheatmap  specifically the color ("direct_gene_based_AUC") it does not show similar results.

e.g. The RSS scores for the eRegulon Atf3_direct_+/+_(239g):

`rss = regulon_specificity_scores(
    scplus_mudata = scplus_mdata,
    variable = "scATAC_counts:celltype",
    modalities = ["direct_gene_based_AUC"]
)
`

`rss.filter(like='Atf3')`

<img width="191" alt="Skærmbillede 2024-05-25 kl  13 14 02" src="https://github.com/aertslab/scenicplus/assets/125036200/39526a98-90ad-4642-b49c-2a4e630f947f">

and based on the average AUC value for the eRegulon Atf3_direct_+/+_(239g):

Code for making the dotplotheatmap;
`heatmap_dotplot(
    scplus_mudata = subset_mdata,
    color_modality = "direct_gene_based_AUC",
    size_modality = "direct_region_based_AUC",
    group_variable = "scRNA_counts:celltype",
    eRegulon_metadata_key = "direct_e_regulon_metadata",
    color_feature_key = "Gene_signature_name",
    size_feature_key = "Region_signature_name",
    feature_name_key = "eRegulon_name",
    sort_data_by = "direct_gene_based_AUC",
    orientation = "vertical",
    figsize = (16, 20)
)`

The color_matrix:

`group_by = scplus_mdata.obs["scRNA_counts:celltype"].tolist()`

`color_features = scplus_mdata.uns["direct_e_regulon_metadata"]["Gene_signature_name"].unique()`

`color_matrix = direct_gene_based_AUC[color_features]`

`color_matrix_avg = color_matrix.groupby(group_by).mean()`

`color_matrix_avg.filter(like='Atf3')`

<img width="195" alt="Skærmbillede 2024-05-25 kl  13 15 48" src="https://github.com/aertslab/scenicplus/assets/125036200/b1049ef5-baee-4ec0-94c2-60deb7058cbf">

Since both are calculated on the same dataframe; scplus_mdata.mod["direct_gene_based_AUC"], am I not quite sure why I get this inconsistency... 

Hope you can help.

Best regards,

Laura

**Comment by SeppeDeWinter** [comment, marked as answer]
Hi @laurajohanne 

Both are different metrics:

The RSS in essence calculates how similar the actual distribution of AUC values for your eRegulon is to what the distribution would be if the regulon is only active in a single cell type.

While the values in the dotplotheatmap are average AUC values per cell type.

I hope this helps?

Best,

Seppe

**Reply by laurajohanne** [reply]
Thank you, that helps. 

Best regards, Laura.

**Comment by YH-Zheng** [comment]
Hi @SeppeDeWinter,

I would like to know how to filter cell type specific eRegulons by RSS value? Is it to select a threshold value for filtering?

Best regards,
Yuhui
